packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -8,12 +8,25 @@
 export default function copyEmptyArrayProps(oldObj, newObj) {
   const newObjWithFixedEmptyArray = Object.entries(oldObj).reduce(
     (acc, [key, value]) => {
-      const replaceWithEmptyArray =
-        Array.isArray(value) && newObj[key] === undefined
-      acc[key] = replaceWithEmptyArray ? [] : newObj[key]
+      const isArray = Array.isArray(value)
+
+      // check if the value is array on `oldObj` and the key not found in `acc` (newObjWithFixedEmptyArray)
+      if (isArray && acc[key] === undefined) {
+        // init `key` with empty array
+        return { ...acc, [key]: [] }
```

Comment:
I like this a lot 👍

Reply:
I also relearned this dynamic property trick from @MarkusAzer 😎

Reply:
this looks to me like a breaking change